rm-portal/src/main/java/org/ow2/proactive_grid_cloud_portal/rm/client/nodesource/serialization/ImportFromCatalogPanel.java

```diff
@@ -0,0 +1,147 @@
+        this.nodeSourceListBox.setWidth("190px");
+        this.nodeSourceListBox.addChangeHandler(onNodeSourceSelected -> requestNodeSourceConfiguration());
+        add(this.nodeSourceListBox);
+    }
+
+    private void requestNodeSourceConfiguration() {
+        String selectedNodeSourceInList = this.nodeSourceListBox.getSelectedValue();
+        if (!selectedNodeSourceInList.equals(SELECT_NODE_SOURCE_GENERIC_ENTRY)) {
+            String nodeSourceConfigurationUrl = this.catalogUrl + "/buckets/node-sources/resources/" +
+                                                selectedNodeSourceInList + "/raw";
+            RequestBuilder request = new RequestBuilder(RequestBuilder.GET, nodeSourceConfigurationUrl);
+            request.setHeader(SESSION_ID_PARAMETER_NAME, LoginModel.getInstance().getSessionId());
+            request.setCallback(new RequestCallback() {
+                @Override
+                public void onResponseReceived(Request request, Response response) {
+                    ImportFromCatalogPanel.this.importNodeSourcePanel.handleNodeSourceImport(response.getText());
+                }
+
+                @Override
+                public void onError(Request request, Throwable t) {
+                    GWT.log("Import node source from catalog failed. Request was " + request.toString(), t);
+                }
+            });
+            try {
+                request.send();
+            } catch (RequestException e) {
+                GWT.log("Request sent to catalog failed", e);
+            }
+        }
+    }
+
+    private void requestNodeSourcesList() {
+        String nodeSourcesListUrl = catalogUrl + "/buckets/node-sources/resources?kind=nodesource";
+        RequestBuilder request = new RequestBuilder(RequestBuilder.GET, nodeSourcesListUrl);
+        request.setHeader(SESSION_ID_PARAMETER_NAME, LoginModel.getInstance().getSessionId());
+        request.setCallback(new RequestCallback() {
+            @Override
+            public void onResponseReceived(Request request, Response response) {
+                JSONArray nodeSources = JSONParser.parseStrict(response.getText()).isArray();
+                for (int i = 0; i < nodeSources.size(); i++) {
+                    JSONObject nodeSource = nodeSources.get(i).isObject();
+                    String nodeSourceName = nodeSource.get(NAME_KEY).isString().stringValue();
+                    ImportFromCatalogPanel.this.nodeSourceListBox.addItem(nodeSourceName);
+                }
+                ImportFromCatalogPanel.this.nodeSourceListBox.setEnabled(true);
+            }
+
+            @Override
+            public void onError(Request request, Throwable exception) {
+                GWT.log("Error occurred when fetching buckets from Catalog");
+            }
+        });
+        try {
+            request.send();
+        } catch (RequestException e) {
+            GWT.log("Error occured when fetching buckets from Catalog");
+            e.printStackTrace();
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Use a logger to log this exception. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1148)